docs/sql-ref-literals.md

```diff
@@ -0,0 +1,506 @@
+---
+layout: global
+title: Literals
+displayTitle: Literals
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+A literal (also known as a constant) represents a fixed data value. Spark SQL supports the following literals:
+
+ * [String Literal](#string-literal)
```

Comment:
We also support binary literals with X-prefixed strings e.g. X'ABCD'

Reply:
I am debating if we should include X-prefixed strings. Seems to me 'ABCD' is a string literal. X applies to this string literal.

Reply:
But, I think we should document this type constructor somewhere, too.

Reply:
It's actually a BinaryType lit

Reply:
I saw DB2 SQL reference document binary string constants separately. I think we need to add this. I will change the string literal syntax. @cloud-fan WDYT?
<img width="816" alt="Screen Shot 2020-04-21 at 3 35 28 PM" src="https://user-images.githubusercontent.com/13592258/79920782-c4291e80-83e5-11ea-83f3-0abbfef8aca3.png">

Reply:
It's binary literal, let's add a new section to introduce it